apps/prairielearn/src/migrations/20240911213026_submission_embedding_store.sql

```diff
@@ -0,0 +1,8 @@
+CREATE TABLE IF NOT EXISTS submission_grading_context_embeddings (
+  id bigserial PRIMARY KEY,
+  embedding vector (1536),
+  submission_id BIGINT NOT NULL REFERENCES submissions ON DELETE CASCADE ON UPDATE CASCADE,
+  submission_text text
+);
+
+CREATE INDEX ON submission_grading_context_embeddings USING hnsw (embedding vector_cosine_ops);
```

Comment:
Do you expect this index to help us in practice? That is, when would Postgres be able to use this index given that we won't ever be querying _every_ submission embedding but rather only querying those for the same assessment question? For that matter, why isn't there an index on `submission_id`?

Reply:
You are absolutely right. We don't need the index on the embeddings, and I created a new index on the submission id.